### PR TITLE
One car has one user + New Auth in backend

### PR DESCRIPTION
**Pull request by LucasBrazi06**

### CODE REVIEW

File: src/server/rest/v1/service/CarService.ts [hunk 1]
```diff
@@ -218,98 +217,76 @@ export default class CarService {
...
-    UtilsService.assertObjectExists(action, carCatalog, `Car Catalog ID '${filteredRequest.carCatalogID}' does not exist`,
-      MODULE_NAME, 'handleCreateCar', req.user);
-    // Keep the current user to add for Basic role
-    const carUserToAdd = filteredRequest.usersAdded.find((carUser) => carUser.user.id === req.user.id);
-    // Check Car
-    const car = await CarStorage.getCarByVinLicensePlate(req.tenant,
-      filteredRequest.licensePlate, filteredRequest.vin, {
-        withUsers: Authorizations.isBasic(req.user) ? true : false,
+    // Only Admin can create Pool Car
+    if (filteredRequest.type === CarType.POOL_CAR && !Authorizations.isAdmin(req.user)) {
```
Comment (mahdibenromdhane) [inline src/server/rest/v1/service/CarService.ts:225]: I think this case should be handled by the authorizations framework

File: src/server/rest/v1/service/CarService.ts [hunk 2]
```diff
@@ -320,31 +297,30 @@ export default class CarService {
       Action.UPDATE, Entity.CAR, MODULE_NAME, 'handleUpdateCar');
     // Filter
     const filteredRequest = CarSecurity.filterCarUpdateRequest(req.body);
-    UtilsService.assertIdIsProvided(action, filteredRequest.id, MODULE_NAME, 'handleUpdateCompany', req.user);
     // Check
     UtilsService.checkIfCarValid(filteredRequest, req);
+    // Only Admin can create Pool Car
+    if (filteredRequest.type === CarType.POOL_CAR && !Authorizations.isAdmin(req.user)) {
```
Comment (mahdibenromdhane) [inline src/server/rest/v1/service/CarService.ts:303]: I think this case should be handled by the authorizations framework

File: src/types/requests/HttpCarRequest.ts [hunk 3]
```diff
@@ -26,32 +25,33 @@ export interface HttpCarRequest extends HttpByIDRequest {
...
+  default: boolean;
 }
 
 export interface HttpCarsRequest extends HttpDatabaseRequest {
   Search: string;
   CarMaker: string;
-  WithUsers: boolean;
+  WithUser: boolean;
   UserID: string;
 }
```
[the remark this answers is not in the document]
Comment (LucasBrazi06) [reply src/types/requests/HttpCarRequest.ts:57]: Just removed it

File: src/server/rest/v1/service/CarService.ts (hunk 2 above)
Comment (LucasBrazi06) [reply src/server/rest/v1/service/CarService.ts:303]: I added new auth named 'CREATE_POOL_CAR' to Admin and Site Admin

File: src/server/rest/v1/service/CarService.ts (hunk 1 above)
Comment (LucasBrazi06) [reply src/server/rest/v1/service/CarService.ts:225]: I added new auth named 'CREATE_POOL_CAR' to Admin and Site Admin

Comment (LucasBrazi06) [reply src/server/rest/v1/service/CarService.ts]: Finally I introduced the Assert authorizations to check the content of an Updated/Created object.

File: src/server/rest/v1/service/CarService.ts (hunk 2 above)
Comment (LucasBrazi06) [reply src/server/rest/v1/service/CarService.ts]: Finally I introduced the Assert authorizations to check the content of an Updated/Created object.

**Comment by LucasBrazi06** [comment]
@mahdibenromdhane You can review this PR, I provided the changes you have requested. Thanks!

**Comment by LucasBrazi06** [comment]
> @mahdibenromdhane You can review this PR, I provided the changes you have requested. Thanks!

@mahdibenromdhane I'll merge this PR as there are some failing unit tests fixed by this dev.
You can nevertheless check it in master-qa.
Thanks!

**Pull request merged**